tensorflow/python/ops/image_grad_deterministic_test.py

```diff
@@ -127,6 +135,164 @@ def resize_bilinear_gradients(local_seed):
           self.assertAllEqual(result_a, result_b)
 
 
+class CropAndResizeOpDeterminismExceptionsTest(test.TestCase):
+  """Test d9m-unimplemented exceptions from CropAndResizeBackprop{Image|Boxes}.
+
+  Test that tf.errors.UnimplementedError is thrown or not thrown, as
+  appropriate, by the GPU code-paths for CropAndResizeBackprop{Image|Boxes} when
+  deterministic ops are enabled.
+
+  This test assumes that the base op test runs all the same test cases when
+  deterministic ops are not enabled and will therefore detect erroneous
+  exception throwing in those cases.
+  """
+
+  def _genParams(self, dtype=dtypes.float32):
+    batch_size = 1; image_height = 10; image_width = 10; channels = 1
+    image_shape = (batch_size, image_height, image_width, channels)
+    num_boxes = 3
+    boxes_shape = (num_boxes, 4)
+    random_seed.set_seed(123)
+    image = random_ops.random_normal(shape=image_shape, dtype=dtype)
+    boxes = random_ops.random_uniform(shape=boxes_shape, dtype=dtypes.float32)
+    box_indices = random_ops.random_uniform(
+        shape=(num_boxes,), minval=0, maxval=batch_size, dtype=dtypes.int32)
+    crop_size = constant_op.constant([3, 3], dtype=dtypes.int32)
+    return image, boxes, box_indices, crop_size
+
+  def _imageErrorMessage(self):
```

Comment:
Make this a class constant:

```python
 _IMAGE_ERROR_MESSAGE = ("Deterministic GPU implementation of" +
                         " CropAndResizeBackpropImage not available")
```

And same for `_boxesErrorMessage`.

Reply:
Thanks for the simplification suggestion.

I originally put these messages in the class-level with the intention of using them in the parent class, but I ended up deciding to not test the error messages in the parent class; there it seemed sufficient to simply check that the correct exception type was being thrown. In the upcoming commit, I have simplified this so that there is nothing at all at the class level related to the exception messages.